documentation/modules/ROOT/pages/security-oidc.adoc

```diff
@@ -0,0 +1,188 @@
+
+Quarkus introduced an experimental *Dev Services For Keycloak* feature which is enabled by default when the `quarkus-oidc` extension is started in dev mode. It starts a Keycloak container and initializes it by registering the existing Keycloak realm or creating a new realm with the client and users for you to start developing your Quarkus application secured by Keycloak immediately.
+
+So let's see how you can start using Bearer Tokens for securing your endpoints.
+
+== Add the OIDC extension
+
+Just open a new terminal window, and make sure you’re at the root of your `{project-name}` project, then run:
+
+[.console-input]
+[source,bash,subs="+macros,+attributes"]
+----
+./mvnw quarkus:add-extension -Dextensions="io.quarkus:quarkus-oidc"
+----
+
+== KeyCloak setup
+Go to http://localhost:8080/q/dev/[Dev UI] and select the OpenId Connect Card linking to a Keycloak page. Click on the http://localhost:8080/q/dev/io.quarkus.quarkus-oidc/provider[Provider: Keycloak link] and you will see a Keycloak page which will be presented slightly differently depending on how Dev Services for Keycloak feature has been configured.
+
+TIP: By default, `alice` and `bob` users (with the passwords matching the names), and user and admin roles are created. `alice` has both admin and user roles and `bob` - the user role.
+
+However, we will add our own user, role and group by following these steps:
+ 
+. Click on http://localhost:55105/auth[KeyCloak Admin link] in the left corner.
+. Login using `admin' as user and password.
+. Go to https://raw.githubusercontent.com/redhat-developer-demos/quarkus-tutorial/master//jwt-token/realm.json and save it on your local machine.
+. Go to http://localhost:55105/auth/admin/master/console/#/realms/quarkus/partial-import[Import] in the KeyCloak Administration console and import `realm.json`. 
+After this step you should have `Subscriber` Role and Group.
+. Go to http://localhost:55105/auth/admin/master/console/#/realms/quarkus/users[Users] and add the `jdoe` user having the name as password. The `jdoe` user should be mapped to the `Subscriber` role and group.
+
+== Modify SecureResource
+
+If you need to access `JsonWebToken` claims, you can simply inject the token itself.
+
+Change the `SecureResource` Java class in `src/main/java` in the `com.redhat.developers` package with the following contents:
+
+[.console-input]
+[source,java]
+----
+package com.redhat.developers;
+
+import org.eclipse.microprofile.jwt.JsonWebToken;
+
+import javax.annotation.security.RolesAllowed;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+@Path("/secure")
+public class SecureResource {
+    
+    @Inject
+    JsonWebToken jwt;
+
+    @GET
+    @Path("/claim")
+    @RolesAllowed("Subscriber") <1>
+    public String getClaim() {
+        return "Access for subject " + jwt.getName() + " is granted";
+    }
+
+}
+----
+<1> The endpoint is accessible only to users that have `Subscriber` role.
+
```

Comment:
Can you add a NOTE/INFO admonition explaining how to configure Quarkus app when using an external Keycloack when DevServices is not used?

Reply:
Also, can you add a TIP explaining how to disable DevServices?

Reply:
Sure!

Reply:
In my latest commits, I added as last sections how to configure your application and run a local KeyCloak server.